eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -375,11 +375,30 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
     }
   }
 
-  override def sentInfo(id: Either[UUID, ByteVector32])(implicit timeout: Timeout): Future[Seq[OutgoingPayment]] = Future {
-    id match {
+  override def sentInfo(id: Either[UUID, ByteVector32])(implicit timeout: Timeout): Future[Seq[OutgoingPayment]] = {
+    val outgoingDbPayments = id match {
```

Comment:
The database call should be done within a `Future`, like before. It forces you to compose futures which is not very elegant but I think it's really important.

Reply:
Good catch, done in https://github.com/ACINQ/eclair/pull/2142/commits/bf8a61a7937712e49b41b2c97494196843c4e60f